src/components/Select.tsx

```diff
@@ -59,14 +63,29 @@ export default function Select<T>({
 
   const selectedOptions = options.filter((option) => isSelected(option.value));
 
+  const hasUsedProperty = options.some((option) => option.used !== undefined);
```

Comment:
What if someone passes used:false, then hasUsedProperty would be true, right? I think we should check `option.used === true` instead?